pilot/cmd/pilot-agent/main.go

```diff
@@ -264,6 +264,9 @@ func initProxy(args []string) (*model.Proxy, error) {
 		proxy.IPAddresses = []string{podIP.String()}
 	}
 
+	// After IP addresses are set, let us discover IPMode.
+	proxy.DiscoverIPMode()
```

Comment:
Why not call **DiscoverIPMode** after obtaining all the IPs from the node in later?

Reply:
That makes sense. I will fix it.